bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,431 @@
+                        String level = tokenizer.poll();
+                        updateGroup(address, level);
+                    } else if ("indicatorkill".equals(command)) {
+                        updateGroup(address, "-1");
+                    } else {
+                        logger.warn("Unhandled trigger command {} - status {}", command, status);
+                    }
+                    break;
+                }
+                case "clock": {
+                    tokenizer.poll();
+                    @Nullable
+                    String address = "";
+                    @Nullable
+                    String value = "";
+                    @Nullable
+                    String type = tokenizer.poll();
+                    if ("date".equals(type)) {
+                        address = tokenizer.poll() + "/1";
+                        value = tokenizer.poll();
+                    } else if ("time".equals(type)) {
+                        address = tokenizer.poll() + "/0";
+                        value = tokenizer.poll();
+                    } else if (!"request_refresh".equals(type)) {
+                        logger.warn("Received unknown clock event: {}", status);
+                    }
+                    if (value != null && !value.isEmpty()) {
+                        updateGroup(address, value);
+                    }
+                    break;
+                }
+                default:
+                    if ("lighting".equals(commentTokenizer.peek())) {
+                        commentTokenizer.poll();
+                        @Nullable
+                        String commentToken = commentTokenizer.peek();
+                        if ("SyncUpdate".equals(commentToken)) {
+                            commentTokenizer.poll();
+                            @Nullable
+                            String address = commentTokenizer.poll();
+
+                            @Nullable
+                            String level = commentTokenizer.poll();
+                            level.replace("level=", "");
+                            updateGroup(address, level);
+                        }
+                    } else {
+                        logger.warn("Received unparsed event: '{}'", status);
+                    }
+                    break;
+            }
+        }
+
+    }
+
+    @NonNullByDefault
+    private class EventMonitor extends EventCallback {
+
+        @Override
+        public boolean acceptEvent(int event_code) {
+            return true;
+        }
+
+        @SuppressWarnings({ "null" })
+        @Override
+        public void processEvent(@Nullable CGateSession cgate_session, int eventCode,
+                @Nullable GregorianCalendar event_time, @Nullable String event) {
+            if (event == null) {
+                return;
+            }
+            LinkedList<String> tokenizer = new LinkedList<String>(Arrays.asList(event.trim().split("\\s+")));
+
+            if (eventCode == 701) {
```

Comment:
To minimize the amount of executed code, you should make sure to only populate variables when you know they will be used.
```suggestion
            if (eventCode == 701) {
                LinkedList<String> tokenizer = new LinkedList<String>(Arrays.asList(event.trim().split("\\s+")));
```

Reply:
Fixed